### PR TITLE
Waku2 key enr part two

### PR DESCRIPTION
**Pull request by jemboh**

### CODE REVIEW

File: src/lib/enr/enr.spec.ts [hunk 1]
```diff
@@ -8,30 +9,46 @@ import { bytesToHex, hexToBytes } from "../utils";
...
-    it("should encodeTxt and decodeTxt", async () => {
+    it.only("should encodeTxt and decodeTxt", async () => {
       const peerId = await PeerId.create({ keyType: "secp256k1" });
       const enr = ENR.createFromPeerId(peerId);
       const keypair = createKeypairFromPeerId(peerId);
       enr.setLocationMultiaddr(new Multiaddr("/ip4/18.223.219.100/udp/9000"));
       enr.multiaddrs = [
         new Multiaddr(
-          "/dns4/node-01.do-ams3.wakuv2.test.statusim.net/tcp/443/wss"
+          "/dns4/node1.do-ams.wakuv2.test.statusim.net/tcp/443/wss"
```
Comment (jemboh) [inline src/lib/enr/enr.spec.ts:27]: this helped reduce the size of the ENR to under 300 bytes although I'm concerned about in the real world where we don't have control over this

File: src/lib/enr/enr.spec.ts [hunk 2]
```diff
@@ -8,30 +9,46 @@ import { bytesToHex, hexToBytes } from "../utils";
...
+
 describe("ENR", function () {
   describe("Txt codec", () => {
-    it("should encodeTxt and decodeTxt", async () => {
+    it.only("should encodeTxt and decodeTxt", async () => {
       const peerId = await PeerId.create({ keyType: "secp256k1" });
       const enr = ENR.createFromPeerId(peerId);
       const keypair = createKeypairFromPeerId(peerId);
       enr.setLocationMultiaddr(new Multiaddr("/ip4/18.223.219.100/udp/9000"));
       enr.multiaddrs = [
```
Comment (jemboh) [inline src/lib/enr/enr.spec.ts:25]: what's also strange is that if you comment out the `multiaddrs` completely, it still throws the `Error: data long segment too short` error!

File: src/lib/enr/enr.spec.ts (hunk 1 above)
Comment (D4nte) [reply src/lib/enr/enr.spec.ts:23]: In the real world then:
- Operator can create several ENRs with different multiaddrs
- We can look into moving away from ENR if it's too limited.

File: src/lib/enr/enr.spec.ts (hunk 2 above)
Comment (D4nte) [reply src/lib/enr/enr.spec.ts:21]: This has been resolved as it was due to a bug with the base64 codec, right?

File: src/lib/discovery/dns.ts [hunk 3]
```diff
@@ -34,14 +41,22 @@ export class DnsNodeDiscovery {
...
+    enrTreeUrls: string[],
+    wantedNodeCapabilityCount: NodeCapabilityCount | undefined
+  ): Promise<ENR[]> {
     let totalSearches = 0;
     const peers: ENR[] = [];
 
     const networkIndex = Math.floor(Math.random() * enrTreeUrls.length);
     const { publicKey, domain } = ENRTree.parseTree(enrTreeUrls[networkIndex]);
 
+    console.log("wantedNodeCapabilityCount:", wantedNodeCapabilityCount);
```
Comment (jemboh) [inline src/lib/discovery/dns.ts:58]: I was stuck in here on how to match the node (peer?) to the `wantedNodeCapabilityCount`. I tried keeping a count but couldn't find an elegant way to keep track of the multiple counts.

File: src/lib/enr/enr.spec.ts (hunk 2 above)
Comment (jemboh) [reply src/lib/enr/enr.spec.ts:21]: Yes!

**Comment by D4nte** [comment]
Blocked by https://github.com/status-im/nwaku/issues/951

**Comment by D4nte** [comment]
bors merge

**Pull request merged**